consul/resource_consul_node.go

```diff
@@ -67,10 +73,20 @@ func resourceConsulNodeCreate(d *schema.ResourceData, meta interface{}) error {
 	address := d.Get("address").(string)
 	name := d.Get("name").(string)
 
+	nodeMeta := make(map[string]string)
+	if v, ok := d.GetOk("node_meta"); ok {
+		for k, j := range v.(map[string]interface{}) {
+			nodeMeta[k] = j.(string)
+		}
+	} else {
+		nodeMeta = nil
+	}
+
 	registration := &consulapi.CatalogRegistration{
 		Address:    address,
 		Datacenter: dc,
 		Node:       name,
+		NodeMeta:   nodeMeta,
```

Comment:
Just alternatively you could assign this after to `registration` instead of doing `else { nodeMeta = nil}`. Somewhat more concise and leaves the pattern open for more attributes in the future.

Reply:
Made a change in 1f440f0. Not sure if this is what you mean.

Sorry, I am new to Go.